ibc-clients/cw-context/src/context/client_ctx.rs

```diff
@@ -0,0 +1,179 @@
+        _client_id: &ClientId,
+        height: &Height,
+    ) -> Result<(Timestamp, Height), ContextError> {
+        let time_key = self.client_update_time_key(height);
+
+        let time_vec = self.retrieve(time_key)?;
+
+        let time = u64::from_be_bytes(time_vec.try_into().expect("invalid timestamp"));
+
+        let timestamp =
+            Timestamp::from_nanoseconds(time).map_err(ClientError::InvalidPacketTimestamp)?;
+
+        let height_key = self.client_update_height_key(height);
+
+        let revision_height_vec = self.retrieve(height_key)?;
+
+        let revision_height =
+            u64::from_be_bytes(revision_height_vec.try_into().expect("invalid height"));
+
+        let height = Height::new(0, revision_height)?;
+
+        Ok((timestamp, height))
+    }
+}
+
+impl<'a, C: ClientType<'a>> ClientExecutionContext for Context<'a, C> {
+    type ClientStateMut = C::ClientState;
+
+    fn store_client_state(
+        &mut self,
+        _client_state_path: ClientStatePath,
+        client_state: Self::ClientStateMut,
+    ) -> Result<(), ContextError> {
+        let prefixed_key = self.prefixed_key(ClientStatePath::leaf());
+
+        let encoded_client_state = self.encode_client_state(client_state)?;
+
+        self.insert(prefixed_key, encoded_client_state);
+
+        Ok(())
+    }
+
+    fn store_consensus_state(
+        &mut self,
+        consensus_state_path: ClientConsensusStatePath,
+        consensus_state: Self::ConsensusStateRef,
+    ) -> Result<(), ContextError> {
+        let prefixed_key = self.prefixed_key(consensus_state_path.leaf());
+
+        let encoded_consensus_state = C::ConsensusState::encode_thru_any(consensus_state);
+
+        let wasm_consensus_state = WasmConsensusState {
+            data: encoded_consensus_state,
+        };
+
+        let encoded_wasm_consensus_state = C::ConsensusState::encode_thru_any(wasm_consensus_state);
+
+        self.insert(prefixed_key, encoded_wasm_consensus_state);
+
+        Ok(())
+    }
+
+    fn delete_consensus_state(
+        &mut self,
+        consensus_state_path: ClientConsensusStatePath,
+    ) -> Result<(), ContextError> {
+        let prefixed_key = self.prefixed_key(consensus_state_path.leaf());
+
+        self.remove(prefixed_key);
+
+        Ok(())
+    }
+
+    fn store_update_meta(
+        &mut self,
+        _client_id: ClientId,
+        height: Height,
+        host_timestamp: Timestamp,
+        host_height: Height,
+    ) -> Result<(), ContextError> {
+        let time_key = self.client_update_time_key(&height);
+
+        let prefixed_time_key = self.prefixed_key(time_key);
+
+        let time_vec: [u8; 8] = host_timestamp.nanoseconds().to_be_bytes();
+
+        self.insert(prefixed_time_key, time_vec);
+
+        let height_key = self.client_update_height_key(&height);
+
+        let prefixed_height_key = self.prefixed_key(height_key);
+
+        let revision_height_vec: [u8; 8] = host_height.revision_height().to_be_bytes();
```

Comment:
Do we need this `: [u8; 8]`? [`u64::to_be_bytes` returns `[u8; 8]`](https://doc.rust-lang.org/std/primitive.u64.html#method.to_be_bytes).

There are a few others.

Reply:
[6de21ed](https://github.com/cosmos/ibc-rs/pull/1167/commits/6de21edb62137b9d74daf1bcacf3b626e45013cc)